### PR TITLE
test(refactored): remove servicemanager

### PR DESCRIPTION
**Pull request by sanakhandvsa**
## Description
Refactored tests by removing dependency and duplication of serviceManager

Related issue: [JIRA_TICKET_NUMBER](LINK_TO_JIRA_TICKET)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php [hunk 1]
```diff
@@ -427,77 +472,89 @@ public function indexAction_WithPost_ShouldRedirectToPage_DependantOnDecision_Pr
...
+            LicenceVehicleManagement::class,
+        ])->andReturn($this->sessionMock);
+
+        $controllerPluginManagerMock->shouldReceive('get')->withArgs([
+            Url::class,
+        ])->andReturn($this->urlHelperMock);
+
+        $container->shouldReceive('get')->withArgs([
+            FormValidator::class,
+        ])->andReturn($this->formValidatorMock);
```
Comment (JoshuaLicense) [inline test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php:536]: Isn't `$container` a local variable? What's going on here?

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]

File: test/Olcs/src/Controller/Auth/LoginControllerTest.php [hunk 2]
```diff
@@ -483,85 +559,67 @@ public function postAction_FailedAuthentication_FlashesAccountDisabledWhenAuthen
...
+        ])->andReturn($this->flashMessengerMock);
+
+        $container->shouldReceive('get')->withArgs([
+            FormHelperService::class,
+        ])->andReturn($this->formHelperMock);
+
+        $controllerPluginManagerMock->shouldReceive('get')->withArgs([
+            Redirect::class,
+        ])->andReturn($this->redirectHelperMock);
+        $this->setUpDefaultServices();
```
[the remark this answers is not in the document]
Comment (sanakhandvsa) [reply test/Olcs/src/Controller/Auth/LoginControllerTest.php:608]: **Consistency with Factory Structure**: The structure of the setUpSut method mirrors the `LoginControllerFactory` invocation method.

[the remark this answers is not in the document]
Comment (JoshuaLicense) [reply test/Olcs/src/Controller/Auth/LoginControllerTest.php:608]: I'm more getting at that this method is not doing anything as it's creating a _local_ variable (`container`, & `controllerPluginManagerMock`), configuring it, and then not using it.

File: test/Olcs/src/Controller/Auth/LoginControllerTest.php [hunk 3]
```diff
@@ -571,63 +629,28 @@ protected function identity(bool $isAnonymous = true)
...
     private function authChallengeContainer()
     {
-        if (!$this->serviceManager->has(AuthChallengeContainer::class)) {
-            $instance = $this->setUpMockService(AuthChallengeContainer::class);
-            $this->serviceManager->setService(AuthChallengeContainer::class, $instance);
-        }
-        $instance = $this->serviceManager->get(AuthChallengeContainer::class);
-        assert($instance instanceof MockInterface);
-        return $instance;
+        assert($this->authChallengeContainerMock instanceof MockInterface);
```
[the remark this answers is not in the document]
Comment (sanakhandvsa) [reply test/Olcs/src/Controller/Auth/LoginControllerTest.php:653]: The addition of the assert statement directly on the `$this->authChallengeContainerMock` was intended to maintain the same level of integrity and assurance in our tests as previously achieved, but in a more direct and simplified manner. The original code fetched an instance from the `serviceManager` and then used an assert to ensure it was of the correct type. My intention was to replicate this assertion to confirm the mock object's type without the intermediary step of retrieving it from the serviceManager.

File: test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php [hunk 4]
```diff
@@ -56,14 +93,16 @@ public function indexAction_IsCallable()
...
      */
     public function indexAction_ReturnsViewModel()
     {
         // Setup
         $this->setUpSut();
         $routeMatch = new RouteMatch([]);
-
+        // Define Expectations
+        $this->flashMessengerMock->shouldReceive('hasMessages')->withAnyArgs();
+        $this->flashMessengerMock->shouldReceive('getMessages')->withAnyArgs();
```
[the remark this answers is not in the document]
Comment (sanakhandvsa) [reply test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php:104]: When we used the ServiceManager, it possibly provided actual instances or lenient mocks (like FlashMessenger) that didn't enforce strict expectations for method calls. After switching to Mockery, we must explicitly define all mock interactions, enhancing test specificity but requiring us to set clear expectations for each method call.

`Received Mockery_3_Laminas_Mvc_Plugin_FlashMessenger_FlashMessenger::hasMessages()`
if not it is giving error no expectations were specified

File: test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php [hunk 5]
```diff
@@ -427,77 +472,89 @@ public function indexAction_WithPost_ShouldRedirectToPage_DependantOnDecision_Pr
...
-            $this->serviceManager->setService(Redirect::class, $instance);
-        }
-        return $this->serviceManager->get(Redirect::class);
-    }
-
-    /**
-     * @return MockInterface|FormValidator
-     */
-    protected function formValidator(): MockInterface
+    protected function formValidator()
```
[the remark this answers is not in the document]
Comment (sanakhandvsa) [reply test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php:544]: - **formValidatorMock** is initialized as a property to provide a mock object that can be utilized across tests.
- Within the **formValidator** method, we're invoking the **isValid** method of the mock object to set expectations specific to that method.

File: test/Olcs/src/Controller/Auth/LoginControllerTest.php [hunk 6]
```diff
@@ -75,8 +106,23 @@ class LoginControllerTest extends MockeryTestCase
...
         self::setupLogger();
+        $this->sut = new LoginController(
+            $this->authenticationAdapterMock,
+            $this->authenticationServiceMock,
+            $this->currentUserMock,
+            $this->flashMessengerMock,
+            $this->formHelperMock,
+            $this->redirectHelperMock,
+            $this->authChallengeContainerMock
+        );
```
[the remark this answers is not in the document]
Comment (sanakhandvsa) [reply test/Olcs/src/Controller/Auth/LoginControllerTest.php:125]: Yes Good suggestion, I will  modify where ever in test file separating initialisation after configuring the injected mocks

File: test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php (hunk 1 above)
Comment (sanakhandvsa) [reply test/Olcs/src/Controller/Licence/Vehicle/SwitchBoardControllerTest.php]: removed

**Review by JoshuaLicense — APPROVED** [review]
Looks good! Thanks @sanakhandvsa!

**Review by fibble — APPROVED** [review]

**Pull request merged**